lib/zendesk_apps_support/validations/manifest.rb

```diff
@@ -139,31 +138,32 @@ def default_locale_error(manifest, package)
         end
 
         def missing_location_error(package)
-          missing_keys_validation_error(['location']) unless package.manifest.location?
+          missing_keys_validation_error(['location']) if package.manifest.location_options.empty?
         end
 
         def invalid_location_error(package)
           errors = []
-          manifest_locations = package.manifest.locations
-          manifest_locations.find do |host, locations|
-            product = Product.find_by(name: host)
-            stub = ZendeskAppsSupport::Manifest::LEGACY_URI_STUB
-            locations_allowed = Location.where(product_code: product.code).map(&:name).push(stub)
-            if (invalid_locations = locations.keys - locations_allowed).any?
-              errors << ValidationError.new(:invalid_location,
-                                  invalid_locations: invalid_locations.join(', '),
-                                  host_name: host,
-                                  count: invalid_locations.length)
+          package.manifest.location_options.each do |location_options|
```

Comment:
thinking of renaming `LocationOptions` to `LocationOptionGroup`, what do you think?